### PR TITLE
Remove middleware warning

### PR DESCRIPTION
**Pull request by penalosa**
## What this PR solves / how to test

Alternative fix to https://github.com/cloudflare/workers-sdk/pull/5999

## Author has addressed the following

- Tests
  - [ ] TODO (before merge)
  - [x] Included
  - [ ] Not necessary because:
- E2E Tests CI Job required? (Use "e2e" label or ask maintainer to run separately)
  - [ ] I don't know
  - [ ] Required / Maybe required
  - [x] Not required because:
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [ ] TODO (before merge)
  - [ ] Included
  - [x] Not necessary because: Change to testing coe
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): 
  - [x] Not necessary because: Not user-facing

### CODE REVIEW

File: packages/wrangler/src/__tests__/middleware.test.ts [hunk 1]
```diff
@@ -270,8 +313,6 @@ describe("middleware", () => {
 	});
 
 	describe("unchanged functionality when wrapping with middleware", () => {
-		runInTempDir();
-
 		process.env.EXPERIMENTAL_MIDDLEWARE = "true";
```
Comment (petebacondarwin) [inline packages/wrangler/src/__tests__/middleware.test.ts:316]: I think we should be able to remove this now?

File: packages/wrangler/src/deployment-bundle/apply-middleware.ts [hunk 2]
```diff
@@ -41,19 +41,22 @@ export async function applyMiddlewareLoaderFacade(
 		`__MIDDLEWARE_${index}__`,
 		path.resolve(getBasePath(), m.path),
 	]);
+	if (process.env.NODE_ENV === "test") {
```
Comment (petebacondarwin) [inline packages/wrangler/src/deployment-bundle/apply-middleware.ts:44]: 👍

**Comment by petebacondarwin** [comment]
Looks like there are some failing tests

File: packages/wrangler/src/deployment-bundle/apply-middleware.ts (hunk 2 above)
Comment (threepointone) [reply packages/wrangler/src/deployment-bundle/apply-middleware.ts:44]: took this idea and made a PR with _only_ this! https://github.com/cloudflare/workers-sdk/pull/6043

**Pull request closed without merging**